c10/core/impl/CUDATraceTLS.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <c10/core/impl/PyInterpreter.h>
+
+namespace c10 {
+namespace impl {
+
+static constexpr char trace_cuda_event_creation_fn_name[] =
+  "CUDAEventCreationCallbacks";
+static constexpr char trace_cuda_event_deletion_fn_name[] =
+  "CUDAEventDeletionCallbacks";
+static constexpr char trace_cuda_event_record_fn_name[] =
+  "CUDAEventRecordCallbacks";
+static constexpr char trace_cuda_event_wait_fn_name[] =
+  "CUDAEventWaitCallbacks";
+static constexpr char trace_cuda_memory_allocation_fn_name[] =
+  "CUDAMemoryAllocationCallbacks";
+static constexpr char trace_cuda_memory_deallocation_fn_name[] =
+  "CUDAMemoryDeallocationCallbacks";
+static constexpr char trace_cuda_stream_allocation_fn_name[] =
+  "CUDAStreamAllocationCallbacks";
+
+struct C10_API CUDATraceTLS {
+  static void set_trace(const PyInterpreter*);
+  static const PyInterpreter* get_trace();
+};
+
+struct C10_API CUDATraceFunctionWrapper {
```

Comment:
Same: this struct appears to only be used in PyInterpreter.h/cpp, can it just be defined over there instead of here?

Reply:
I've moved it as well.